tests/phpunit/tests/test-switch-blog-custom-code.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+if ( is_multisite() ) :
+
+	/**
+	 * Test class for custom code in multisite.
+	 */
+	class Switch_Blog_Custom_Code_Test extends PLL_Multisites_TestCase {
+		/**
+		 * Posts by blogs, stores only posts in default language if applicable.
+		 *
+		 * @var WP_Post[]
+		 */
+		private $posts = array();
+
+		/**
+		 * Creates posts for a given blog.
+		 *
+		 * @param WP_Site        $blog      Current site object.
+		 * @param PLL_Admin|null $pll_admin Polylang admin object, null if deactivated.
+		 * @param array          $languages Array of blog languages data, empty if none.
+		 * @return void
+		 */
+		protected function create_fixtures_for_blog( WP_Site $blog, $pll_admin = null, array $languages = array() ) {
+			$post = $this->factory()->post->create_and_get();
+
+			$this->posts[ (int) $blog->blog_id ] = $post;
+
+			if ( $pll_admin instanceof PLL_Admin && ! empty( $languages ) ) {
+				// Current blog has Polylang activated, let's create multilingual fixtures.
+				$pll_admin->model->post->set_language( $post->ID, $languages[0]['slug'] );
+				$tr_post = $this->factory()->post->create_and_get();
+				$pll_admin->model->post->save_translations(
+					$post->ID,
+					array(
+						$languages[0]['slug'] => $post->ID,
+						$languages[1]['slug'] => $tr_post->ID,
+					)
+				);
+			}
+		}
+
+		public function test_switch_blog_with_custom_code() {
+			$pll_frontend = $this->get_pll_frontend_env();
+			$pll_frontend->curlang = $pll_frontend->model->get_language( 'en' ); // English is a common language for blogs with Polylang activated, @see {PLL_Multisites_TestCase::set_up()}.
```

Comment:
Not at all. 
I needed a common language across blogs, and given that I've coded `PLL_Multisites_TestCase` I know English is fitting. So I added a comment for clarity, meaning I didn't chose a random language.
Also, I wouldn't trust a LLM for multisite related code.

Reply:
> Also, I wouldn't trust a LLM for multisite related code.

I wouldn't even trust a human 😏

Reply:
To clarify my 1st comment, it was a joke because the sentence "English is a common language for blogs with Polylang activated" sounded like it comes from statistics about sites that use Polylang (in general, out of the context of this test) 😅

Reply:
I see 🤣 
I admit it sounds a bit artificial^^